### ISSUE TITLE
sba_impl configuration invalid

### ISSUE THREAD

**Issue by hermilohdez** [Kitware/TeleSculptor#215]
**kwiver_fmv_set_1 example with ubuntu:

After successfully completed the maptk_track_features solution,  maptk_bundle_adjust_tracks stops after next:**

2016-11-28 08:20:54.631 WARN algorithm.cxx(237): Configuration Failure: invalid option
2016-11-28 08:20:54.631 WARN algorithm.cxx(237):    sba_impl:type = ceres
2016-11-28 08:20:54.631 WARN algorithm.cxx(237):    valid options are
2016-11-28 08:20:54.631 WARN algorithm.cxx(237):       hierarchical
2016-11-28 08:20:54.631 WARN algorithm.cxx(237):       vxl
2016-11-28 08:20:54.631 DEBUG hierarchical_bundle_adjust.cxx(258): Config Check Fail: sba_impl configuration invalid.
2016-11-28 08:20:54.631 WARN algorithm.cxx(247): Configuration Failure Backtrace: bundle_adjuster:hierarchical
2016-11-28 08:20:54.631 ERROR bundle_adjust_tracks.cxx(274): Config Check Fail: Failed config check in bundle_adjuster algorithm.
2016-11-28 08:20:54.631 ERROR bundle_adjust_tracks.cxx(703): Configuration not valid.

**Comment by mleotta** [comment]
What version of MAP-Tk are you using?  Or what branch/commit if not a release version?

**Comment by hermilohdez** [comment]
MAP-Tk 0.9.0

**Comment by mleotta** [comment]
I'll see if I can reproduce it.  It's possible that those example config files have become outdated and need updating.

**Comment by mleotta** [comment]
I was not able to reproduce the issue.  Looking closer at your posted log message it looks like you don't have the Ceres Solver plugin.  This example configuration uses the Ceres bundle adjuster nested inside a hierarchical bundle adjuster.  It doesn't seem to be a plugin path issue because it found at least the "core" and "vxl" plugins.  Are you using the installer package or did you build it yourself?  If you built it, did you enable Ceres in CMake?

Note that I did find another unrelated issue when running this example.  Part way through the run I get a core dump from within the Ceres plugin.  I'll look into fixing that as well.

**Comment by hermilohdez** [comment]
I am using the installer package. Do I still need the Ceres plugin?

**Comment by hermilohdez** [comment]
Next is the link from where I got it:

https://github.com/Kitware/maptk/releases

**Comment by mleotta** [comment]
The installer package should provide Ceres.  Does your install have `lib/libmaptk_ceres.so` and `lib/maptk/maptk_ceres_plugin.so`?  Maybe there is another reason it is not loading.  Can attach you entire run log?  Can you run the tool as shown below and upload the log.txt?

    maptk_bundle_adjust_tracks -c maptk_bundle_adjust.conf > log.txt 2>&1

Also, what version of Ubuntu are you using?

**Comment by hermilohdez** [comment]
My install have both: lib/libmaptk_ceres.so  and lib/maptk/maptk_ceres_plugin.so. 
[log.txt](https://github.com/Kitware/maptk/files/625846/log.txt)
I'm using Ubuntu 14.04.5 LTS

**Comment by mleotta** [comment]
Thanks.  The error appears to be

    Failed to open module library /home/pk/MAP-Tk-0.9.0-Linux-x86_64/bin/../lib/maptk/maptk_ceres_plugin.so (error: libf77blas.so.3: cannot open shared object file: No such file or directory)

Try this:

    apt-get install libatlas3-base

**Comment by hermilohdez** [comment]
Thanks. Now I ran into another issue:
[log.txt](https://github.com/Kitware/maptk/files/625974/log.txt)

**Comment by mleotta** [comment]
Yep, that's the error I found when trying to reproduce your original issue.  I'm working on it.

**Comment by mleotta** [comment]
The error message reported in your log:

    problem_impl.cc:65 Check failed: it != parameter_map.end() Parameter block not found

has now been fixed on the `release` branch.  It will be included in MAP-Tk 0.9.1 and beyond.  However, we aren't ready to release that version just yet.  The bundle adjust tool now completes on that data again, but the results are not as good as before.  The configuration files need further updating.

**Comment by hermilohdez** [comment]
Thank you for let it me know.  MAP-Tk 0.9.1  will have the configuration files updated or there are other issues too? In the meantime, I would try the release branch to become familiar with the code.
